app/model/BaseRule.scala

```diff
@@ -14,11 +29,119 @@ trait BaseRule {
 }
 
 object BaseRule {
-  def toJson(rule: BaseRule): JsObject = Json.obj(
-    "id" -> rule.id,
-    "category" -> rule.category,
-    "description" -> rule.description,
-    "suggestions" -> rule.suggestions,
-    "replacement" -> rule.replacement
-  )
+  implicit val writes: Writes[BaseRule] = Json.writes[BaseRule]
```

Comment:
We convert our rules to a sealed trait here to allow implicit serialisation entities that inherit the `BaseRule` trait – see [the Scala docs](https://www.playframework.com/documentation/2.8.x/ScalaJsonAutomated#Requirements) for why this is needed.